tests/unit_tests/external_interaction/nexus/test_compare_nexus_to_gda_exhaustively.py

```diff
@@ -0,0 +1,266 @@
+        "gda item y_pixels not present",
+    ),
+    "/entry/instrument/detector/detectorSpecific/y_pixels_in_detector": (
+        dtype("uint32"),
+        dtype("int32"),
+    ),
+    "/entry/instrument/detector/module/data_origin": (dtype("uint32"), dtype("int32")),
+    "/entry/instrument/detector/module/data_size": (dtype("uint32"), dtype("int32")),
+    "/entry/instrument/detector/module/data_stride": (dtype("uint32"), dtype("int32")),
+}
+
+
+def mockitem(name, item):
+    m = MagicMock()
+    m.dtype = f"{name} item {item} not present"
+    return m
+
+
+def test_determine_datatype_differences(
+    files_and_groups: FilesAndgroups,
+):
+    gda_nexus, gda_groups, hyperion_nexus, hyperion_groups = files_and_groups
+    diffs = {}
+    for item in gda_groups:
+        # we checked separately if all expected items should be there
+        # but we should probably still check the excepted ones here??
+        if item in hyperion_groups:
+            hyperion_group = hyperion_nexus[item]
+            gda_group = gda_nexus[item]
+            print(hyperion_group, gda_group)
+            assert isinstance(hyperion_group, Group) and isinstance(gda_group, Group)
+            for dset_or_attr in hyperion_group:
+                hyperion_item = mockitem("hyperion", dset_or_attr)
+                gda_item = mockitem("gda", dset_or_attr)
+                try:
+                    hyperion_item = hyperion_group[dset_or_attr]
+                except KeyError:
+                    ...  # should probably correlate this with some key table
+                try:
+                    gda_item = gda_group[dset_or_attr]
+                except KeyError:
+                    ...  # should probably correlate this with some key table
+                if not isinstance(hyperion_item, Group) and not isinstance(
+                    hyperion_item, Datatype
+                ):
+                    assert not isinstance(gda_item, Group) and not isinstance(
+                        gda_item, Datatype
+                    )
+
+                    hyperion_dtype = hyperion_item.dtype
+                    gda_dtype = gda_item.dtype
+                    print(
+                        item,
+                        dset_or_attr,
+                        hyperion_dtype,
+                        gda_dtype,
+                        hyperion_dtype == gda_dtype,
+                    )
+                    if not hyperion_dtype == gda_dtype:
+                        diffs[item + "/" + str(dset_or_attr)] = (
+                            hyperion_dtype,
+                            gda_dtype,
+                        )
+    print(diffs)
+
+
+def test_hyperion_vs_gda_datatypes(
+    files_and_groups: FilesAndgroups,
+):
+    gda_nexus, gda_groups, hyperion_nexus, hyperion_groups = files_and_groups
+    for item in gda_groups:
+        # we checked separately if all expected items should be there
+        # but we should probably still check the excepted ones here??
```

Comment:
If the gda file really is from GDA and we don't update it to a new baseline over time I think the number of things that are not in the original GDA file is going to make this test increasingly less useful, unless we explicitly check the exceptions here.